addons/account_peppol/wizard/peppol_registration.py

```diff
@@ -0,0 +1,266 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models, modules
+from odoo.exceptions import UserError, ValidationError
+
+from odoo.addons.account_peppol.tools.demo_utils import handle_demo
+
+
+class PeppolRegistration(models.TransientModel):
+    _name = 'peppol.registration'
+    _description = "Peppol Registration"
+
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        required=True,
+        default=lambda self: self.env.company,
+    )
+    account_peppol_contact_email = fields.Char(related='company_id.account_peppol_contact_email', readonly=False)
+    account_peppol_edi_mode = fields.Selection(
+        string='EDI mode',
+        selection=[('demo', 'Demo'), ('test', 'Test'), ('prod', 'Live')],
+        compute='_compute_account_peppol_edi_mode',
+        inverse='_inverse_account_peppol_edi_mode',
+        readonly=False,
+    )
+    account_peppol_edi_user = fields.Many2one(
+        comodel_name='account_edi_proxy_client.user',
+        string='EDI user',
+        compute='_compute_account_peppol_edi_user',
+    )
+    account_peppol_migration_key = fields.Char(related='company_id.account_peppol_migration_key', readonly=False)
+    account_peppol_mode_constraint = fields.Selection(
+        selection=[('demo', 'Demo'), ('test', 'Test'), ('prod', 'Live')],
+        compute='_compute_account_peppol_mode_constraint',
+        help="Using the config params, this field specifies which edi modes may be selected from the UI"
+    )
+    account_peppol_phone_number = fields.Char(related='company_id.account_peppol_phone_number', readonly=False)
+    account_peppol_proxy_state = fields.Selection(related='company_id.account_peppol_proxy_state', readonly=False)
+    account_peppol_verification_code = fields.Char(related='account_peppol_edi_user.peppol_verification_code', readonly=False)
+    peppol_eas = fields.Selection(related='company_id.peppol_eas', readonly=False)
+    peppol_endpoint = fields.Char(related='company_id.peppol_endpoint', readonly=False)
+    peppol_warnings = fields.Json(
+        string="Peppol warnings",
+        compute="_compute_peppol_warnings",
+    )
+    smp_registration = fields.Boolean('Register as a receiver')
+
+    # -------------------------------------------------------------------------
+    # ONCHANGE METHODS
+    # -------------------------------------------------------------------------
+
+    @api.onchange('peppol_endpoint')
+    def _onchange_peppol_endpoint(self):
+        for wizard in self:
+            if wizard.peppol_endpoint:
+                wizard.peppol_endpoint = ''.join(char for char in wizard.peppol_endpoint if char.isalnum())
+
+    # -------------------------------------------------------------------------
+    # COMPUTE METHODS
+    # -------------------------------------------------------------------------
+
+    @api.depends("company_id.account_edi_proxy_client_ids")
```

Comment:
but then this will only be computed when the company_id is changed, right? we need to recompute when there is a change in `account_edi_proxy_client_ids`